src/schema/course/utils.ts

```diff
@@ -35,3 +38,30 @@ export const buildCourseQuery = (params: QueryCourseListArgs) => {
     { key: 'audience_max_age_gt', value: params.audienceMaxAgeGt },
   ]);
 };
+
+export const getCoursesList = async ({
+  dataSources,
+  include,
+  ids,
+}: {
+  dataSources: any;
+  include: string[];
+  ids: string[];
+}) => {
+  const courses = await Promise.all(
+    ids.map(async (id) => {
+      try {
+        const query = buildCourseQuery({ include });
+        const course = await dataSources.courseAPI.getCourseDetails(id, query);
+        return normalizeKeys(course);
+      } catch (e) {
```

Comment:
It seems that some of the resolver methods handle exception and others dont. Also some of handling is located lower level as in here. I would move all handling to the resolver-level, including messaging to sentry. Resolvers could even use some sort of middleware or other unified way of handling errors, perhaps apollo server has some specific way to do it? Perhaps this is bigger topic and worth of another ticket :)

Reply:
apollo way is to throw errors in resolvers, it already handles them

Reply:
Need to investigate if we can return data and throw error at the same time in the resolver. Sounds strange anyway...

Reply:
https://www.apollographql.com/docs/apollo-server/data/errors/

can't see how we should handle those errors, because that resolver might have error in one of the events fetched by ids but others work fine and we can't just throw error in the resolver?

Reply:
It just seems that _some_ errors are reported to the sentry and some are not, which seems to be a bit inconsistent to me.

Reply:
Well yeah that is valid issue :) should improve that.